lib/flags.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+class Flags {
+  static from(...args) {
+    const values = new Map(args.map((v, i) => [v, i]));
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
```

Comment:
Also, I think it is important to implement support for construction of the `FlagsClass` object from `Uint64` number (since that's how it is going to be stored).
At the moment that is only achievable via changing the `value` field of the resulting object, which isn't very convenient.